### DISCUSSION TITLE
Error rendering a page with TeX

### DISCUSSION THREAD

**Discussion by YconquestY** [vuepress-theme-hope/vuepress-theme-hope#2128] [Q&A] [answered]
Hi folks, VuePress fails in rendering a page of my blog which contains TeX equations. Below is the log. Can someone help me determine where the problem lies?
```
01:59:57 [vite] Internal server error: Error parsing JavaScript expression: Expecting Unicode escape sequence \uXXXX. (1:2)
  Plugin: vite:vue
  File: D:/Blog/docs/.vuepress/.temp/pages/nerf/nerf_ndc.html.vue
  144|  \mathbf{I}_{3 \times 3} &amp; \begin{matrix} -l \\ -b \\ -n&#x27; \end{matrix} \\
  145|  \boldsymbol{0}_{1 \times 3} &amp; 1
  146|  \end{bmatrix}}_{{\text{move corner to origin}}} \\
     |                  ^
  147|  &amp;=
  148|  \begin{bmatrix}
      at createCompilerError (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:19:19)
      at processExpression (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:3486:25)
      at Array.transformExpression (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:3319:24)
      at traverseNode (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:2162:41)
      at traverseChildren (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:2153:9)
      at traverseNode (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:2204:13)
      at traverseChildren (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:2153:9)
      at traverseNode (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:2204:13)
      at traverseChildren (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:2153:9)
      at traverseNode (D:\Blog\node_modules\@vue\compiler-core\dist\compiler-core.cjs.js:2204:13)
```

Furthermore, the output of `vuepress info` is
```
$ npm run test
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

> blog@1.0.0 test
> vuepress info


  System:
    OS: Windows 10 10.0.22000
    CPU: (16) x64 AMD Ryzen 7 5800H with Radeon Graphics
    Memory: 2.53 GB / 13.86 GB
  Binaries:
    Node: 16.16.0 - D:\Software\Node.js\node.EXE
    Yarn: Not Found
    npm: 8.11.0 - D:\Software\Node.js\npm.CMD
  Utilities:
    Git: 2.37.0.
  Browsers:
    Chrome: Not Found
    Edge: Spartan (44.22000.120.0), Chromium (104.0.1293.54)
  npmPackages:
    @vuepress/bundler-vite:  2.0.0-beta.49
    @vuepress/bundler-webpack: Not Found
    @vuepress/cli:  2.0.0-beta.49
    @vuepress/client: ^2.0.0-beta.49 => 2.0.0-beta.49
    @vuepress/core:  2.0.0-beta.49
    @vuepress/markdown:  2.0.0-beta.49
    @vuepress/plugin-active-header-links:  2.0.0-beta.49
    @vuepress/plugin-back-to-top:  2.0.0-beta.49
    @vuepress/plugin-container:  2.0.0-beta.49
    @vuepress/plugin-docsearch: Not Found
    @vuepress/plugin-external-link-icon:  2.0.0-beta.49
    @vuepress/plugin-git:  2.0.0-beta.49
    @vuepress/plugin-google-analytics: Not Found
    @vuepress/plugin-medium-zoom:  2.0.0-beta.49
    @vuepress/plugin-nprogress:  2.0.0-beta.49
    @vuepress/plugin-palette:  2.0.0-beta.49
    @vuepress/plugin-prismjs:  2.0.0-beta.49
    @vuepress/plugin-pwa: Not Found
    @vuepress/plugin-pwa-popup: Not Found
    @vuepress/plugin-register-components: Not Found
    @vuepress/plugin-search: Not Found
    @vuepress/plugin-shiki: Not Found
    @vuepress/plugin-theme-data:  2.0.0-beta.49
    @vuepress/plugin-toc: Not Found
    @vuepress/shared:  2.0.0-beta.49
    @vuepress/theme-default:  2.0.0-beta.49
    @vuepress/utils:  2.0.0-beta.49
    vue: ^3.2.36 => 3.2.37
    vue-loader: Not Found
    vue-router:  4.1.3
    vuepress: ^2.0.0-beta.49 => 2.0.0-beta.49
    vuepress-vite:  2.0.0-beta.49
    vuepress-webpack: Not Found
```

Thanks.

**Comment by Mister-Hope** [comment, marked as answer]
No, you need to provide a minimal reproduction. the log already show you the error file in line 3.

paste your file and config in an empty template and provide me

**Reply by YconquestY** [reply]
Sure, the minimal reproduction is [here](https://stackblitz.com/edit/vuepress-scrkty?file=docs/nerf/nerf_ndc.md). The error is weird since there were no problems when I used other KaTeX plugins, and I'm pretty sure the equations are formulated correctly.

**Reply by Mister-Hope** [reply]
Emm, I have some issues installing deps on stackblitz these days, can you provide a repo?

I am not sure this belongs to my network service provider or my own problem, but I can not even trigger download action while getting tons of 404.

![image](https://user-images.githubusercontent.com/33315834/184666029-5c83073f-0677-4983-b0e0-570f60d8e733.png)

**Reply by YconquestY** [reply]
Alright, see necessary files in [this repo](https://github.com/YconquestY/tmp.git).

**Reply by YconquestY** [reply]
Good evening. Any update on this issue? Is it because of syntax errors in my `md` or a problem with the theme?

Thanks.

**Reply by Mister-Hope** [reply]
Sorry, but unfoturnately, my laptop cpu fan is down. I didn't discover it at the first time, and my laptop is overheated then force shuting down. And this caused my battery having a great deformation. I managed to open the laptop and remove the battery, but the shell has  a little bit deformation because of battery and  I am also worried about the SSD next to the battery.

I may not get my laptop run in a few days, I will try to see if it can still work after sending it to repair center to get  a report. And I may buy a new device because it's already 5 years old.

So I probably won't push any commits to  github in the  next few days.

**Reply by Mister-Hope** [reply]
You can try to set output to only html, to see if it's caused by annonation, if so, that could be an workaround.

**Reply by YconquestY** [reply]
Nevermind, I appreciate your help.

I like your theme :)

And btw, may I know how to customize font?

**Reply by Mister-Hope** [reply]
Just noticing you, I just struggled out from https://github.com/vuepress-theme-hope/vuepress-theme-hope/pull/2153. I will review your issue in 24 hours, and probably a new version will be landed in 48h

**Reply by Mister-Hope** [reply]
I will relook into this soon

**Reply by Mister-Hope** [reply]
Hi there, the reproduction is gone. My question is can you show a standard tex syntax containing `{{xxx}}`

`_{{\text{move corner to origin}}}` This is not a standard tex syntax as you should use `_{\text{move corner to origin}}`

**Reply by Mister-Hope** [reply]
If a standard tex can't have `{{ xxx }}` then you are expected to escape or avoid them yourself.

**Reply by YconquestY** [reply]
I get it now. Thank you.
